src/test/java/core/basesyntax/operations/BalanceOperationTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.operations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.operations.exception.InvalidInputDataException;
+import core.basesyntax.service.ReportCreator;
+import core.basesyntax.service.impl.ReportCreatorImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class BalanceOperationTest {
+    private BalanceOperation balanceOperation;
+    private ReportCreator reportCreator;
+
+    @BeforeEach
+    void setUp() {
+        balanceOperation = new BalanceOperation();
+        reportCreator = new ReportCreatorImpl();
+    }
+
+    @Test
+    void handle_BalanceOperation_Ok() {
+        String emptyReportExpected = "fruit,quantity";
+        String emptyReport = reportCreator.createReport();
```

Comment:
dont use report creator in Operation tests. Just test data state in storage. Report creating shold be testet by its own test
same for other operation handler tests